src/hotspot/cpu/aarch64/frame_aarch64.cpp

```diff
@@ -499,7 +509,13 @@ frame frame::sender_raw(RegisterMap* map) const {
 
   // Must be native-compiled frame, i.e. the marshaling code for native
   // methods that exists in the core system.
-  return frame(sender_sp(), link(), sender_pc());
+
+  // Native code may or may not have signed the return address when saving it
+  // to the stack. In addition, we do not know which key was used to sign it.
+  // Therefore, all we can do is strip it.
+  address sender_pc = pauth_strip_pointer(this->sender_pc());
```

Comment:
Would it not make more sense to strip the `sender_pc()` ? Don't all users expect it to be stripped?

Reply:
> Would it not make more sense to strip the `sender_pc()` ? Don't all users expect it to be stripped?

The current code is being fairly cautious:

Native frames: saved PC might be signed. Therefore strip the pc.

Interpreter frames and compiled frames: the PC should never be signed. Assert this is the case.

For both Interpreter and Native frames we are getting the pc via sender_pc(), so can't push the strip inside sender_pc().

Alternatively, we could always strip instead of asserting. But that's not quite as nice.

Reply:
Mmm, but you very easily fix that by defining `raw_sender_pc()`. Most users don't want to be bothered by this stuff, and having `sender_pc()` return an encrypted PC is, frankly, maintainer abuse. Anyone who wants the `raw_sender_pc()` should ask for it.

Reply:
Ok, that way makes sense. Updated as suggested.